packages/x-charts-premium/src/BarChartPremium/BarChartPremium.tsx

```diff
@@ -9,20 +9,43 @@ import { ChartsTooltip } from '@mui/x-charts/ChartsTooltip';
 import { ChartsWrapper } from '@mui/x-charts/ChartsWrapper';
 import { ChartsSurface } from '@mui/x-charts/ChartsSurface';
 import { ChartsGrid } from '@mui/x-charts/ChartsGrid';
-import { BarPlot } from '@mui/x-charts/BarChart';
+import { BarPlot, type BarSeries } from '@mui/x-charts/BarChart';
 import { ChartsOverlay } from '@mui/x-charts/ChartsOverlay';
 import { ChartsAxisHighlight } from '@mui/x-charts/ChartsAxisHighlight';
 import { ChartsAxis } from '@mui/x-charts/ChartsAxis';
 import { ChartZoomSlider } from '@mui/x-charts-pro/ChartZoomSlider';
 import { ChartsBrushOverlay } from '@mui/x-charts/ChartsBrushOverlay';
 import { ChartsClipPath } from '@mui/x-charts/ChartsClipPath';
-import { useChartContainerProProps } from '@mui/x-charts-pro/internals';
+import { seriesPreviewPlotMap, useChartContainerProProps } from '@mui/x-charts-pro/internals';
 import type { BarChartPremiumPluginSignatures } from './BarChartPremium.plugins';
 import { useBarChartPremiumProps } from './useBarChartPremiumProps';
 import { BAR_CHART_PREMIUM_PLUGINS } from './BarChartPremium.plugins';
 import { ChartDataProviderPremium } from '../ChartDataProviderPremium';
+import { type BarItemIdentifier, type RangeBarSeriesType } from '../models';
+import { RangeBarPlot } from './RangeBar/RangeBarPlot';
+import { RangeBarPreviewPlot } from '../ChartZoomSlider/internals/previews/RangeBarPreviewPlot';
 
-export interface BarChartPremiumProps extends BarChartProProps {}
+seriesPreviewPlotMap.set('rangeBar', RangeBarPreviewPlot);
```

Comment:
The zoom preview is pro, but the range bar is premium, so the zoom preview doesn't know that there is a range bar preview plot. 

I'm using this method to register the preview plot for the range bar from the premium package. Let me know if you see any downsides or have any suggestion.

Reply:
Done in https://github.com/mui/mui-x/pull/20716